test/parallel/test-wasm-web-api.js

```diff
@@ -0,0 +1,181 @@
+'use strict';
+
+const common = require('../common');
+const fixtures = require('../common/fixtures');
+
+const assert = require('assert');
+const events = require('events');
+const { createServer } = require('http');
+
+assert.strictEqual(typeof WebAssembly.compileStreaming, 'function');
+assert.strictEqual(typeof WebAssembly.instantiateStreaming, 'function');
+
+const simpleWasmBytes = fixtures.readSync('simple.wasm');
+
+// Sets up an HTTP server with the given response handler and calls fetch() to
+// obtain a Response from the newly created server.
+async function testRequest(handler) {
+  const server = createServer((_, res) => handler(res)).unref().listen(0);
+  await events.once(server, 'listening');
+  const { port } = server.address();
+  return fetch(`http://127.0.0.1:${port}/`);
+}
+
+// Runs the given function both with the promise itself and as a continuation
+// of the promise. We use this to test that the API accepts not just a Response
+// but also a Promise that resolves to a Response.
+function withPromiseAndResolved(makePromise, consume) {
+  return Promise.all([
+    consume(makePromise()),
+    makePromise().then(consume),
+  ]);
+}
+
+function testCompileStreaming(responseCallback) {
+  return withPromiseAndResolved(
+    () => testRequest(responseCallback),
+    common.mustCall((response) => {
+      return WebAssembly.compileStreaming(response);
+    }, 2)
+  );
+}
+
+function testCompileStreamingRejection(responseCallback, rejection) {
+  return withPromiseAndResolved(
+    () => testRequest(responseCallback),
+    common.mustCall((response) => {
+      return assert.rejects(() => {
+        return WebAssembly.compileStreaming(response);
+      }, rejection);
+    }, 2)
+  );
+}
+
+(async () => {
```

Comment:
maybe an interesting test to include would be using `new Response()` with a file stream?

Reply:
I didn't expect file streams to emit `ArrayBuffer`s (instead of `ArrayBufferView`s). Updated the implementation to accept both and added a test.